src/utils/resizeShadowCameraFrustum.js

```diff
@@ -0,0 +1,76 @@
+const boundingBox = new THREE.Box3();
+const frustumBox = new THREE.Box3();
+const inverseLightMatrixWorld = new THREE.Matrix4();
+const tempBox = new THREE.Box3();
+const FRUSTUM_PADDING = 1;
+const NEAR_CLIPPING_PLANE = -500;
+const FAR_CLIPPING_PLANE = 500;
+const verts = [
+  new THREE.Vector3(),
+  new THREE.Vector3(),
+  new THREE.Vector3(),
+  new THREE.Vector3(),
+  new THREE.Vector3(),
+  new THREE.Vector3(),
+  new THREE.Vector3(),
+  new THREE.Vector3()
+];
+
+export default function resizeShadowCameraFrustum(light, scene) {
```

Comment:
Hmm, seems a bit scary to be doing this every frame for each (probably only one) directional light. Its very likely the environment is going to be the largest thing, so the frustum actually wont end up changing anyway... Maybe just do it on environment load, or when objects are added/removed from the scene... Both of these could be a bit buggy for a really small environment with objects moving outside its bounds.. hmm

Reply:
I can just disable it on mobile for now and we can tune it as we learn more about our shadow requirements.